pandas/io/packers.py

```diff
@@ -160,14 +160,14 @@ def writer(fh):
 def read_msgpack(path_or_buf, encoding='utf-8', iterator=False, **kwargs):
     """
     Load msgpack pandas object from the specified
-    file path
+    file path.
 
     THIS IS AN EXPERIMENTAL LIBRARY and the storage format
     may not be stable until a future release.
 
     Parameters
     ----------
-    path_or_buf : string File path, BytesIO like or string
+    path_or_buf : str, path object, URL or file-like object
```

Comment:
can u be more explicit about path object; essentially say what you are doing below when you reference LocalPath

Reply:
do this everywhere you say path object

Reply:
What about *"If you want to pass in a path object, pandas accepts either pathlib.Path or py._path.local.LocalPath."* as in [pd.read_csv](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html)? 

Actually, we could first agree upon a paragraph per type, then place blocks in each `read_*`, in the same order as the types appear in `path_or_buf `. For instance:

1. `str, URL`: Any valid string path is acceptable. The string could be a URL. Valid URL schemes include http, ftp, s3, and file. For file URLs, a host is expected. A local file could be: file://localhost/path/to/table.csv.
2. `path object`: If you want to pass in a path object, pandas accepts either pathlib.Path or py._path.local.LocalPath.
3. `file-like object`: By file-like object, we refer to objects with a read() method, such as a file handler (e.g. via builtin open function) or StringIO.

Thus a `path_or_buf : str, path object, URL or file-like object` would have (1), (2) and (3) in the description.

Reply:
yes that is the ideal to have identially phrases; a way to do this would be to define these phrases in pandas.core.generic, in a shared_docs entry; then simply compose them.

Reply:
Done this in 6a3704bccbcb34e51513c2d0836b028ef5f2f4bc 1864c183ddb0a410e64d26019e4f9d9d9f572840 and 2ec85f9f5d2f101ec71a249352dd7122bc741c84, just by replicating the same description over and over

Reply:
@jreback can you check my comment above?